internal/rootcoord/meta_table.go

```diff
@@ -705,22 +706,35 @@ func (mt *MetaTable) RenameCollection(ctx context.Context, dbName string, oldNam
 	log := log.Ctx(ctx).With(
 		zap.String("db", dbName),
 		zap.String("oldName", oldName),
+		zap.String("newDBName", newDBName),
 		zap.String("newName", newName),
 	)
+
 	if dbName == "" {
 		log.Warn("db name is empty")
 		dbName = util.DefaultDBName
 	}
 
+	if newDBName == "" {
+		log.Warn("target db name is empty")
+		newDBName = dbName
+	}
+
+	// check target db
+	targetDB, ok := mt.dbName2Meta[newDBName]
+	if !ok {
+		return fmt.Errorf("target database:%s not found", newDBName)
+	}
+
 	//old collection should not be an alias
-	_, ok := mt.aliases.get(dbName, oldName)
+	_, ok = mt.aliases.get(dbName, oldName)
 	if ok {
 		log.Warn("unsupported use a alias to rename collection")
 		return fmt.Errorf("unsupported use an alias to rename collection, alias:%s", oldName)
 	}
 
 	// check new collection already exists
-	newColl, err := mt.getCollectionByNameInternal(ctx, dbName, newName, ts)
+	newColl, err := mt.getCollectionByNameInternal(ctx, newDBName, newName, ts)
```

Comment:
If the user rename a collection without changing the db, this check will fail.

Reply:
Ignore this, in fact I just thinked about the case "rename a collection without changing the db name and collection name".
However, by the way, how do we treat this case? I didn't notice the logic handing this case here.

Reply:
Yes, no matter the collection whether changing the db name, the target collection name should not be in target db

Reply:
So, what about my second question?
It seems strange that we report this error, "duplicated new collection name :%s with other collection name or alias".